pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -1587,7 +1587,11 @@ func (cc *XController) deleteQueueJob(obj interface{}) {
 		accessor.SetDeletionTimestamp(&current_ts)
 	}
 	klog.V(3).Infof("[Informer-deleteQJ] %s enqueue deletion, deletion ts = %v", qj.Name, qj.GetDeletionTimestamp())
-	cc.enqueue(qj)
+	//Remove stale copy
+	cc.eventQueue.Delete(qj)
+	cc.qjqueue.Delete(qj)
+	//Add fresh copy
+	cc.eventQueue.Add(qj)
```

Comment:
Why not use the `enqueue` function here?

Reply:
enqueue calls `cc.eventQueue.Add(qj)` which adds AWs only if not present but a stale AW will be present when user delete's appwrapper.

Reply:
This change makes sense.

Reply:
My understanding is that the `Add` implementation updates the existing object if present, and that performing an `Delete` + `Add` will actually push the item further down the queue, while a call to `enqueue` / `Add` would update the item in-place.

Reply:
I just meant for the line 1594. I understand the previous lines are not present in `enqueue`. Github should have only highlighted line 1594 and no the whole block. The `enqueue` function has some logic to check for errors and stuff so I thought it would be useful here. But in the end the `Add` function is what matters, so no biggie.